src/main/java/net/sf/jabref/gui/OpenRemoteDatabaseDialog.java

```diff
@@ -0,0 +1,322 @@
+
+                    //JPasswordField.getPassword() does not return a String, but a char array.
+                    dbSynchronizer.setUp(DBConnector.getNewConnection(selectedType, hostField.getText(), port,
+                                    databaseField.getText(), userField.getText(), new String(passwordField.getPassword())),
+                                    selectedType, databaseField.getText());
+                    dbSynchronizer.initializeDatabases();
+                    frame.addTab(bibDatabaseContext, true);
+
+                    setGlobalPrefs();
+
+                    frame.output(Localization.lang("Remote_connection_to_%0_server_stablished.", selectedType.toString()));
+                    dispose();
+                } catch (ClassNotFoundException exception) {
+                    JOptionPane.showMessageDialog(OpenRemoteDatabaseDialog.this, exception.getMessage(), Localization.lang("Driver error"),
+                            JOptionPane.ERROR_MESSAGE);
+                } catch (SQLException exception) {
+                    JOptionPane.showMessageDialog(OpenRemoteDatabaseDialog.this, exception.getMessage(),
+                            Localization.lang("Connection error"), JOptionPane.ERROR_MESSAGE);
+                } catch (Exception exception) {
+                    JOptionPane.showMessageDialog(OpenRemoteDatabaseDialog.this, exception.getMessage(),
+                            Localization.lang("Warning"), JOptionPane.WARNING_MESSAGE);
+                }
+            }
+        };
+        connectButton.addActionListener(openAction);
+
+        Action cancelAction = new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                dispose();
+            }
+        };
+        cancelButton.addActionListener(cancelAction);
+
+        /**
+         * Set up a listener which updates the default port number once the selection in dbTypeDropDown has changed.
+         */
+        Action dbTypeDropDownAction = new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                portField.setText(
+                        Integer.toString(DBConnector.getDefaultPort((DBType) dbTypeDropDown.getSelectedItem())));
+            }
+        };
+        dbTypeDropDown.addActionListener(dbTypeDropDownAction);
+
+        // Add enter button action listener
+        connectButton.getInputMap(JComponent.WHEN_IN_FOCUSED_WINDOW).put(KeyStroke.getKeyStroke(KeyEvent.VK_ENTER, 0),
+                "Enter_pressed");
+        connectButton.getActionMap().put("Enter_pressed", openAction);
+
+    }
+
+    /**
+     * Fetches possibly saved data and configures the control elements respectively.
+     */
+    private void applyGlobalPrefs() {
+
+        String remoteDatabaseType = Globals.prefs.get(REMOTE_DATABASE_TYPE);
+        if (remoteDatabaseType != null) {
+            if (remoteDatabaseType.equals(DBType.ORACLE.toString())) {
```

Comment:
Move this code to `DBType` so that you can call `dbTypeDropDown.setSelectedItem(DBType.fromString(remoteDatabaseType))`